crates/nu-command/src/filters/default.rs

```diff
@@ -146,6 +147,17 @@ fn default(
             && matches!(input, PipelineData::Value(ref value, _) if value.is_empty()))
     {
         Ok(value.into_pipeline_data())
+    } else if let PipelineData::ListStream(ls, metadata) = input {
+        let span = ls.span();
+        let mut stream = ls.into_inner().peekable();
+        if stream.peek().is_none() {
+            return Ok(value.into_pipeline_data());
```

Comment:
We only need to do the empty ListStream replacement when `--empty` is passed (as represented by `default_when_empty`)